### ISSUE TITLE
Error instaning WebView

### ISSUE THREAD

**Issue by donhuvy** [SteveSandersonMS/WebWindow#133]
![image](https://user-images.githubusercontent.com/1328316/103495691-3d7e7980-4e6e-11eb-9f28-2afccf26947b.png)

**Comment by donhuvy** [comment]
[BlazorDesktopApp.zip](https://github.com/SteveSandersonMS/WebWindow/files/5762724/BlazorDesktopApp.zip)

**Comment by donhuvy** [comment]
```
System.AccessViolationException: 'Attempted to read or write protected memory. This is often an indication that other memory is corrupt.'
```

**Comment by joshdavidson613** [comment]
donhuvy - I am having the EXACT same problem.

**Comment by donhuvy** [comment]
I think about convert ASP.NET Core Blazor WebAssembly (all of Client project, Server project, Shared project) to Desktop app. I think a better WebView, we can run `dotnet run` in Server project source code, then run it. No need WebView for Desktop, use web browser, still no need Server PC.

**Comment by havenindustries** [comment]
I am having the same problem as well, did any of you find a solution?

**Comment by joshdavidson613** [comment]
No, not fixed. It has to do with something in the CEF libraries, so it is pretty deep in there.

**Comment by TheCakeMonster** [comment]
I've managed to get past this problem by installing the latest version of Edge from the Dev channel - version 89.0.767.0. However, it only raises a separate problem - the error message "The specified procedure could not be found".

If I open the WebWindow.Native solution it won't build with an error on line 34 of WebWindow.Windows.cpp; this seems to suggest that one of the method signatures has changed.

I notice that the WebView2 runtime is now officially released - for Windows at the very least - so I'm wondering if it's now possible to update WebWindow.Native to make use of that somehow for a more stable solution.

**Comment by ericnyc646** [comment]
I have the same issue as everyone else.

**Comment by TheCakeMonster** [comment]
I think Mobile Blazor Bindings is the logical successor to this experiment, and if I understand it correctly it will become part of .NET MAUI in .NET 6. I'm trying out MBB in .NET Core 3.1 and it's going really well. There are a couple of missing bits of functionality, but very few, and I've encountered very little by way of issues.

It might be worth you having a look at that.

https://github.com/dotnet/MobileBlazorBindings

Have a go with the hybrid app template, which creates a working solution that shows that whilst you can use XAML, you can also/instead use standard web razor syntax - so you have the choice of whichever best suits you.

**Comment by TheCakeMonster** [comment]
It's all covered in the comprehensive documentation, but it's worth calling out that if you use the HTML UI then you need to have a dev version of Microsoft Edge installed for it to show up - I think that's true even now in Feb 2021, long after the official release of the installable WebView package.

**Comment by dcuccia** [comment]
> It's all covered in the comprehensive documentation, but it's worth calling out that if you use the HTML UI in Mobile Blazor Bindings then you need to have a dev version of Microsoft Edge installed for it to show up - I think that's true even now in Feb 2021, long after the official release of the installable WebView package.

@TheCakeMonster there's also an "evergreen" standalone installer:

https://developer.microsoft.com/en-us/microsoft-edge/webview2/#download-section

More info here: https://docs.microsoft.com/en-us/microsoft-edge/webview2/concepts/distribution

**Comment by TheCakeMonster** [comment]
@dcuccia OK, thanks, I'll check that out again then. I thought I'd already got that installed and yet still found that I needed a canary version of Edge to have anything show up in the app, but maybe I'm wrong.

**Comment by dcuccia** [comment]
Think it worked for me, but lmk how it works out.

**Comment by TheCakeMonster** [comment]
@dcuccia Yep, you're absolutely right. I think I must have uninstalled the WebView2 runtime after this sample app didn't work with it. Now that I've reinstalled it and uninstalled the dev version of Microsoft Edge, my Mobile Blazor Bindings application still works correctly - on Windows x64 anyway.

Thanks!

I've added a correction notice to my previous comment in an attempt to minimise confusion for those who come after us :-)

**Comment by fingers10** [comment]
any updates on this? I'm also facing this error not able to run blazor desktop sample. Please assist.

**Comment by BennyBread** [comment]
I'm also having the same problem.. Has anybody managed to run the WebWindow?

**Comment by julienGrd** [comment]
> I'm also having the same problem.. Has anybody managed to run the WebWindow?

This project dont make sense now with .net Maui. It should bé closed. Take a look at .net Maui it should fit your needs

**Comment by mhrastegari** [comment]
> > I'm also having the same problem.. Has anybody managed to run the WebWindow?
> 
> This project dont make sense now with .net Maui. It should bé closed. Take a look at .net Maui it should fit your needs

Nonsense :)
Imagine you're creating an app for ecommerce. You don't care about using native UI.
It's enough to have just Blazor for doing all UI and make it run on all 3 Desktop Platforms.

Plus who said MAUI supports WPF, mac cocoa and GTK?!

**Comment by derekantrican** [comment]
I also got this with the sample HelloWorldApp as it is currently. Just cloned the repo, opened WebWindow.Samples.sln, started it up and hit this error. Definitely needs to be fixed. I definitely have chromium-based Edge installed, so not sure what else is failing (or if it would be simple to fallback to something else like old Edge).

**Comment by julienGrd** [comment]
> > > I'm also having the same problem.. Has anybody managed to run the WebWindow?
> > 
> > 
> > This project dont make sense now with .net Maui. It should bé closed. Take a look at .net Maui it should fit your needs
> 
> Nonsense :) Imagine you're creating an app for ecommerce. You don't care about using native UI. It's enough to have just Blazor for doing all UI and make it run on all 3 Desktop Platforms.
> 
> Plus who said MAUI supports WPF, mac cocoa and GTK?!

I dont know why you speak about all of that, anyway this project is not suppose to bé used now since there is other alternative. Its not my point of view, its litteraly what Steve Sanderson Say on the main page of this project. 
By the way have a cross platform app which just run a blazor app inside without Care about thé UI is just what i made for a client in maui and it work perfectly (for my needs, i cant speak for all type of project)
